### DISCUSSION TITLE
Release v0.5.1

### DISCUSSION THREAD

**Discussion by chas-iot** [sqm-autorate/sqm-autorate#149] [Announcements]
**Bug-fix**
- fix for excessive CPU utilisation #139

**Setup changes** #140 
- Watermarks _VERSION if not installed from testing/lua-threads to aid in testing  #146 
- Stops _sqm-autorate_ to avoid starting multiple instances 
- Calls a separate `/usr/lib/sqm-autorate/configure.sh` to guide users with initial configuration

**Configure** #140 
- `/usr/lib/sqm-autorate/configure.sh`
- Requires `/etc/config/sqm-autorate` to be in place to update settings / get previous settings
- Can be run many times
- Attempts to detects the wan upload device and associated ifb4 or veth device if no previous settings available #127 
- Applies rules of thumb to calculate minimum speeds from the chosen 'maximum' speeds

**Comment by richb-hanover** [comment]
This looks like great stuff. Which branch should I use to install it?

**Update:** Answered my own question: `testing/lua-threads` branch. I'll test by following the steps of the README in that branch.

**Comment by richb-hanover** [comment]
Hmmm... the script in `/usr/lib/sqm-autorate/sqm-autorate.lua` still shows _VERSION of 0.5.0. Is this correct?

**Reply by chas-iot** [reply]
> the script in `/usr/lib/sqm-autorate/sqm-autorate.lua` still shows _VERSION of 0.5.0. Is this correct?

Not correct, now fixed
